app/src/androidTest/java/org/mozilla/fenix/ui/BookmarksTest.kt

```diff
@@ -344,4 +416,76 @@ class BookmarksTest {
             verifyFolderTitle("3")
         }
     }
+
+    @Test
+    fun changeBookmarkParentFolderTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        browserScreen {
+            createBookmark(defaultWebPage.url)
+        }.openThreeDotMenu {
+        }.openLibrary {
+        }.openBookmarks {
+        }.openThreeDotMenu(defaultWebPage.url) {
+        }.clickEdit {
+            verifyEditBookmarksView()
+            changeBookmarkTitle(testBookmark.title)
+            saveEditBookmark()
+
+            bookmarksListIdlingResource =
+                RecyclerViewIdlingResource(activityTestRule.activity.findViewById(R.id.bookmark_list))
+            IdlingRegistry.getInstance().register(bookmarksListIdlingResource!!)
+
+            createFolder(bookmarksFolderName)
+
+            IdlingRegistry.getInstance().unregister(bookmarksListIdlingResource!!)
+
+        }.openThreeDotMenu(testBookmark.title) {
+        }.clickEdit {
+            clickParentFolderSelector()
+            selectFolder(bookmarksFolderName)
+            navigateUp()
+            saveEditBookmark()
+
+            IdlingRegistry.getInstance().register(bookmarksListIdlingResource!!)
+
+            selectFolder(bookmarksFolderName)
+            verifyBookmarkedURL(defaultWebPage.url.toString())
+
+            IdlingRegistry.getInstance().unregister(bookmarksListIdlingResource!!)
+        }
+    }
+
+    @Test
+    fun navigateBookmarksFoldersTest() {
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openBookmarks {
+            createFolder("1")
+            getInstrumentation().waitForIdleSync()
+            selectFolder("1")
+            createFolder("2")
+            getInstrumentation().waitForIdleSync()
+            selectFolder("2")
+            verifyCurrentFolderTitle("2")
+            navigateUp()
+            verifyCurrentFolderTitle("1")
+            mDevice.pressBack()
+            verifyBookmarksMenuView()
+        }
+    }
+
+    @Test
+    fun cantSelectDesktopFoldersTest() {
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openBookmarks {
+            bookmarksListIdlingResource =
+                RecyclerViewIdlingResource(activityTestRule.activity.findViewById(R.id.bookmark_list))
+            IdlingRegistry.getInstance().register(bookmarksListIdlingResource!!)
+
```

Comment:
Is this needed in this test? Where is the bookmarks coming from in this test?

Reply:
This is the new implementation, desktop bookmark folders will appear by default in the Bookmarks menu, even when empty. So I thought to wait until the folder loads.